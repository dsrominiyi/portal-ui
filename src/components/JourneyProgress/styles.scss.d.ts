export const journeyWrapper: string;
export const step: string;
export const position: string;
export const active: string;
