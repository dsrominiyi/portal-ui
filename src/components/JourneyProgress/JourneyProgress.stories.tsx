import React from 'react';
import { storiesOf } from '@storybook/react';
import JourneyProgress, { Step, Props } from '.';

const stories = storiesOf('JourneyProgress', module);

export const render = (steps: Step[], activeStep: number, onClick?: Props['onClick']) => () => (
  <JourneyProgress steps={steps} activeStep={activeStep} onClick={onClick} />
);

export const storyRenders = {
  withFiveSteps: render(
    [
      { title: 'Step One', url: '/step/1' },
      { title: 'Step Two', url: '/step/2' },
      { title: 'Step Three', url: '/step/3' },
      { title: 'Step Four', url: '/step/4' },
      { title: 'Step Five', url: '/step/5' }
    ],
    3
  ),
  withTwoSteps: render(
    [
      { title: 'Step One', url: '/step/1' },
      { title: 'Step Two', url: '/step/2' }
    ],
    0
  ),
  withLongTitles: render(
    [
      { title: 'Vehicle Details', url: '/step/1' },
      { title: 'Driver Details', url: '/step/2' },
      { title: 'Policy Details', url: '/step/3' },
      { title: 'Disestablishmentarianism', url: '/step/4' },
      { title: 'Convictions', url: '/step/5' }
    ],
    0
  ),
  withNoSteps: render([], 0)
};

stories.add('With five steps', storyRenders.withFiveSteps);
stories.add('With two steps', storyRenders.withTwoSteps);
stories.add('With long titles', storyRenders.withLongTitles);
stories.add('With no steps', storyRenders.withNoSteps);
