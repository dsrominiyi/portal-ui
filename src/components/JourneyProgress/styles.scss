@import '../../style/_vars';

.journey-wrapper {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;

  a,
  a:hover {
    text-decoration: none;
  }

  .step {
    position: relative;
    background-color: #5ba50a;
    flex: 6rem;
    padding: 0 0 0 35px;
    margin: 0 10px 20px 0;
    vertical-align: middle;
    height: 70px;
    max-width: 220px;
    display: flex;
    justify-content: center; /* align horizontal */
    align-items: center; /* align vertical */

    span {
      display: inline-block;
      max-width: 180px;
      vertical-align: middle;
      text-align: center;
      color: #fff;
      z-index: 2;

      &.position {
        background: #0e2340;
        font-size: 1em;
        font-weight: 300;
        border-radius: 50%;
        color: #fff;
        display: inline-block;
        line-height: 1.6em;
        margin-right: 5px;
        text-align: center;
        width: 1.6em;
        position: absolute;
        top: -15px;
      }
    }

    &::after {
      content: "";
      border-top: 35px solid transparent;
      border-bottom: 35px solid transparent;
      border-left: 35px solid #5ba50a;
      position: absolute;
      right: -35px;
      top: 0;
      z-index: 1;
    }

    &::before {
      content: "";
      border-top: 35px solid transparent;
      border-bottom: 35px solid transparent;
      border-left: 35px solid #fff;
      position: absolute;
      left: 0;
      top: 0;
    }

    &.active,
    &:hover {
      background-color: #97d600;

      &::after {
        border-left: 35px solid #97d600;
      }
    }
  }
}
