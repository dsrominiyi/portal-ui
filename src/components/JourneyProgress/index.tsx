import React, { FunctionComponent } from 'react';

import classNames from 'classnames/bind';
import styles from './styles.scss';

const cx = classNames.bind(styles);

export interface Step {
  title: string;
  url: string;
}

export interface Props {
  steps: Step[];
  activeStep: number;
  onClick?: (stepIndex: number) => void;
}

export const JourneyProgress: FunctionComponent<Props> = ({
  steps,
  activeStep,
  onClick
}: Props) => (
  <div className={styles.journeyWrapper}>
    {steps.map((step: Step, index) => (
      <a
        key={index}
        className={cx({ step: true, active: activeStep === index })}
        href={onClick ? undefined : step.url}
        onClick={onClick && (() => onClick(index))}
        title={step.title}
      >
        <span className={styles.position}>{index + 1}</span>
        <span>{step.title}</span>
      </a>
    ))}
  </div>
);

JourneyProgress.displayName = 'JourneyProgress';
export default JourneyProgress;
