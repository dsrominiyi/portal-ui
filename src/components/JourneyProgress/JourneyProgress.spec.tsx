import { shallow } from 'enzyme';
import { render, storyRenders } from './JourneyProgress.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('JourneyProgress', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });

  test('onClick function is triggered when chevron is clicked', () => {
    const onClick = jest.fn();
    const component = shallow(
      render(
        [
          { title: 'Step One', url: '/step/1' },
          { title: 'Step Two', url: '/step/2' },
          { title: 'Step Three', url: '/step/3' }
        ],
        0,
        onClick
      )()
    );
    const buttons = component.find('.step');

    buttons.forEach((button, index) => {
      button.simulate('click');

      expect(onClick).toHaveBeenCalledWith(index);
    });
  });
});
