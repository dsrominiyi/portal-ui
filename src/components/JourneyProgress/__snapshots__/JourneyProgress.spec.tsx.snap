// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyProgress component story renders match snapshots: withFiveSteps 1`] = `
<div
  className="journeyWrapper"
>
  <a
    className="step"
    href="/step/1"
    key="0"
    title="Step One"
  >
    <span
      className="position"
    >
      1
    </span>
    <span>
      Step One
    </span>
  </a>
  <a
    className="step"
    href="/step/2"
    key="1"
    title="Step Two"
  >
    <span
      className="position"
    >
      2
    </span>
    <span>
      Step Two
    </span>
  </a>
  <a
    className="step"
    href="/step/3"
    key="2"
    title="Step Three"
  >
    <span
      className="position"
    >
      3
    </span>
    <span>
      Step Three
    </span>
  </a>
  <a
    className="step active"
    href="/step/4"
    key="3"
    title="Step Four"
  >
    <span
      className="position"
    >
      4
    </span>
    <span>
      Step Four
    </span>
  </a>
  <a
    className="step"
    href="/step/5"
    key="4"
    title="Step Five"
  >
    <span
      className="position"
    >
      5
    </span>
    <span>
      Step Five
    </span>
  </a>
</div>
`;

exports[`JourneyProgress component story renders match snapshots: withLongTitles 1`] = `
<div
  className="journeyWrapper"
>
  <a
    className="step active"
    href="/step/1"
    key="0"
    title="Vehicle Details"
  >
    <span
      className="position"
    >
      1
    </span>
    <span>
      Vehicle Details
    </span>
  </a>
  <a
    className="step"
    href="/step/2"
    key="1"
    title="Driver Details"
  >
    <span
      className="position"
    >
      2
    </span>
    <span>
      Driver Details
    </span>
  </a>
  <a
    className="step"
    href="/step/3"
    key="2"
    title="Policy Details"
  >
    <span
      className="position"
    >
      3
    </span>
    <span>
      Policy Details
    </span>
  </a>
  <a
    className="step"
    href="/step/4"
    key="3"
    title="Disestablishmentarianism"
  >
    <span
      className="position"
    >
      4
    </span>
    <span>
      Disestablishmentarianism
    </span>
  </a>
  <a
    className="step"
    href="/step/5"
    key="4"
    title="Convictions"
  >
    <span
      className="position"
    >
      5
    </span>
    <span>
      Convictions
    </span>
  </a>
</div>
`;

exports[`JourneyProgress component story renders match snapshots: withNoSteps 1`] = `
<div
  className="journeyWrapper"
/>
`;

exports[`JourneyProgress component story renders match snapshots: withTwoSteps 1`] = `
<div
  className="journeyWrapper"
>
  <a
    className="step active"
    href="/step/1"
    key="0"
    title="Step One"
  >
    <span
      className="position"
    >
      1
    </span>
    <span>
      Step One
    </span>
  </a>
  <a
    className="step"
    href="/step/2"
    key="1"
    title="Step Two"
  >
    <span
      className="position"
    >
      2
    </span>
    <span>
      Step Two
    </span>
  </a>
</div>
`;
